plugins/discovery/discovery_test.go

```diff
@@ -453,6 +457,10 @@ func TestReconfigure(t *testing.T) {
 
 	disco.oneShot(ctx, download.Update{Bundle: updatedBundle})
 
+	if disco.status.Type != "snapshot" {
```

Comment:
Nit: If we could verify that `disco.status` is not nil and then do this check would be better.